common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java

```diff
@@ -119,18 +125,18 @@ public TransportResponseHandler(TransportConf conf, Channel channel) {
     if (checkFetchTimeout) {
       fetchCheckerScheduleFuture =
           fetchTimeoutChecker.scheduleWithFixedDelay(
-              () -> failExpiredFetchRequest(),
+              this::failExpiredFetchRequest,
               fetchTimeoutCheckerInterval,
               fetchTimeoutCheckerInterval,
               TimeUnit.MILLISECONDS);
     }
+
+    registerMetrics();
   }
 
   public void failExpiredPushRequest() {
     long currentTime = System.currentTimeMillis();
-    Iterator<Map.Entry<Long, PushRequestInfo>> iter = outstandingPushes.entrySet().iterator();
-    while (iter.hasNext()) {
-      Map.Entry<Long, PushRequestInfo> entry = iter.next();
+    for (Map.Entry<Long, PushRequestInfo> entry : outstandingPushes.entrySet()) {
```

Comment:
I think we can't rewrite this way, because `removePushRequest` will remove from `outstandingPushes`, which will causes Exception. Using iterator will not cause such problems.